### PR TITLE
Small changes for the systemd service files

### PR DESCRIPTION
**Pull request by vpodzime**
This PR was triggered by the PR #3172 and my [Fedora-packaging activities](https://copr.fedorainfracloud.org/coprs/vpodzime/CFEngine/). The Fedora community RPM should not contain unit files for daemons that are not part of the community package so they cannot be required.

### CODE REVIEW

File: misc/systemd/cfengine3.service.in [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
 After=syslog.target
 
+# Ensure synchronous start behavior
+After=cf-serverd.service
+After=cf-execd.service
+After=cf-monitord.service
+After=cf-postgres.service
+After=cf-apache.service
+After=cf-runalerts.service
+After=cf-hub.service
```
Comment (nickanderson) [inline misc/systemd/cfengine3.service.in]: Oh, that makes more sense.

**Comment by vpodzime** [comment]
@nickanderson changed the service to use `Before=` for sub-services as discussed. Now `systemctl stop cfengine3` is synchronous, start is async and parallel with dependencies specified in the sub-services (e.g. *hub* after *postgres*). Still looks good?

**Comment by vpodzime** [comment]
I consider the lack of complains here as an ACK. We discussed this enough, I think.

**Comment by vpodzime** [comment]
3.12.x version of this PR: https://github.com/cfengine/core/pull/3195

I'm not sure about 3.10.x, but I'm inclined to **not** port this change to that branch due to its potential to change some behavior we are not aware of.

**Pull request merged**